File: src/main/java/com/zuminX/annotations/swagger/ResponseHeader.java
package com.zuminX.annotations.swagger;

import com.zuminX.annotations.AnnotationAttr;
import com.zuminX.annotations.AnnotationStr;
import com.zuminX.names.ClassName;
import com.zuminX.names.SwaggerAnnotation;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * 对应io.swagger.annotations.ResponseHeader
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class ResponseHeader extends AnnotationStr {

  @AnnotationAttr(show = true)
  private String name;

  @AnnotationAttr
  private String description;

  @AnnotationAttr(show = true)
  private ClassName response;

  @AnnotationAttr
  private String responseContainer;

  @Override
  public ClassName getClassName() {
    return SwaggerAnnotation.RESPONSE_HEADER;
  }
}

File: src/main/java/com/zuminX/annotations/swagger/ResponseHeader.kt
package com.zuminX.annotations.swagger

import com.zuminX.annotations.AnnotationAttr
import com.zuminX.annotations.AnnotationStr
import com.zuminX.names.ClassName
import com.zuminX.names.annotation.RESPONSE_HEADER

/**
 * 对应io.swagger.annotations.ResponseHeader
 */
class ResponseHeader : AnnotationStr() {

  @field:AnnotationAttr(show = true)
  var name: String? = null

  @field:AnnotationAttr
  var description: String? = null

  @field:AnnotationAttr(show = true)
  var response: ClassName? = null

  @field:AnnotationAttr
  var responseContainer: String? = null
  
  override fun getClassName(): ClassName = RESPONSE_HEADER
}